webhooks/cluster_test.go

```diff
@@ -155,7 +168,17 @@ func TestClusterValidation(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			g := NewWithT(t)
-			webhook := Cluster{}
+
+			// Sets up the fakeClient for the test case.
+			fakeClient := fake.NewClientBuilder().
```

Comment:
nit/ more out of curiosity: Do we need it here? Looks to me like there is no Cluster referencing a ClusterClass (and more importantly in those cases the webhook should work even when no ClusterClass exists)

Reply:
Not required - I put it this way to give the same setup for the client on each of the tests - not sure we really 'support' a Cluster validation webhook without a client. I've taken it out for now though and added a clearer comment about why we're adding it where we are.